src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageAssets.cs

```diff
@@ -1103,6 +1108,17 @@ private void WriteTransitiveProjectReferences()
                 }
             }
 
+            private void WriteFrameworkReferences()
+            {
+                foreach (var library in _runtimeTarget.Libraries)
```

Comment:
Let's add a DisableTransitiveFrameworkReferences here to match the DisableTransitiveProjectReferences. Also, I think we should dedupe here like we do for framework assemblies. Better to keep the cache small than to dedupe in targets afterwards.

Finally, if transitive framework references are disabled, we should change resolve framework references to not download all packs.

Reply:
@nguerrera I'm not sure there's value in opting out of transitive FrameworkReferences.  There's no split between compile and runtime assets for framework references, so disabling transitive framework references would either have no effect (since you already have the same direct references), or would break your app at runtime (since the framework your dependency needs isn't written to the runtimeconfig).

What do you think?

Reply:
I'd still like there to be an escape hatch.